cmake/pcl_find_avx.cmake

```diff
@@ -5,6 +5,12 @@ function(PCL_CHECK_FOR_AVX)
 
   include(CheckCXXSourceRuns)
   
+  if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG)
+    # Setting -march & -mtune just as required flags for check_cxx_source_runs,
+    # and CMAKE_REQUIRED_FLAGS will be restored after test runs.
+    set(CMAKE_REQUIRED_FLAGS "-march=native -mtune=native")
+  endif()
```

Comment:
Why set them even for the check_cxx_source_runs?

Reply:
`CMAKE_REQUIRED_FLAGS` was needed to tell the CMake which flags to use for test code compiling.

And `-march=native -mtune=native` will cause the compiler to generate the most optimizing code that matches the processor if possible.(e.g. if AVX supported, then compiler will use it)

I made `check_cxx_source_runs` test compiling with these options, hence the AVX/AVX2 code could compile and run well in those supported CPU platforms. And if the test code can't be compiled or run correctly with such optimizing flags, AVX or AVX2 was definitely unsupported.

Reply:
Thanks for the explanation. In that case, we might need to cache the CMAKE_REQUIRED_FLAGS in a variable and then restore from there (instead of clearing CMAKE_REQUIRED_FLAGS)

Reply:
Thank you for your advice!

I think CMAKE_REQUIRED_FLAGS is such a "volatile" variable that it needs to be set to particular value (sometimes this flag leaves blank) before different functions calls including `check_cxx_source_runs`, `check_c_source_runs`, etc. And the flag variable is only for these compile or run checking fields, and won't be used for other ways (I only find the CMAKE_REQUIRED_* in those functions in official docs).

I would prefer to use set/unset method for CMAKE_REQUIRED_FLAGS.